test/base/MarketsStateGetMarketTest.js

```diff
@@ -0,0 +1,108 @@
+// JS Libraries
+const withData = require('leche').withData;
+const { t, createMocks } = require('../utils/consts');
+
+// Mock contracts
+const Mock = artifacts.require("./mock/util/Mock.sol");
+
+// Smart contracts
+const MarketsState = artifacts.require("./base/MarketsState.sol");
+
+const actions = { Supply: 'Supply', Borrow: 'Borrow', Repay: 'Repay' };
+
+contract('MarketsStateGetMarketTest', function (accounts) {
+    const owner = accounts[0];
+    let mocks;
+    let instance;
+    
+    beforeEach('Setup for each test', async () => {
+        instance = await MarketsState.new();
+        instance.addWhitelisted(owner, { from: owner});
+        mocks = await createMocks(Mock, 10);
+    });
+
+    const newAmount = (amount, type, borrowedIndex, collateralIndex) => ({amount, type, borrowedIndex, collateralIndex});
+    const newExpected = (totalSupplied, totalRepaid, totalBorrowed) => ({totalSupplied, totalRepaid, totalBorrowed});
+
+    withData({
+        // (500 borrow - 100 repay) / 2000 Supply = 0.2
+        _1_scenario: [
+            [
+                newAmount(1000, actions.Supply, 0, 1),
+                newAmount(1000, actions.Supply, 0, 1),
+                newAmount(500, actions.Borrow, 0, 1),
+                newAmount(100, actions.Repay, 0, 1),
+            ], 0, 1, newExpected(2000, 100, 500)
+        ],
+        // (2000 borrow - 1000 repay) / 2000 Supply = 0.5
+        _2_scenario: [
+            [
+                newAmount(1000, actions.Supply, 0, 1),
+                newAmount(1000, actions.Supply, 0, 1),
+                newAmount(500, actions.Borrow, 0, 1),
+                newAmount(500, actions.Repay, 0, 1),
+                newAmount(500, actions.Repay, 0, 1),
```

Comment:
should we be able to repay more than what's borrowed? First repay is ok, but second without borrowed amount should be allowed or rejected?

Reply:
Fixed